### PR TITLE
feat(strict-ts): apply to common

### PR DESCRIPTION
**Pull request by capJavert**
This is the one of the PRs in a series to bring our TypeScript types to comply with [strict](https://www.typescriptlang.org/tsconfig/#strict) config option. This will allow us better type checking during builds and development and allow us to ship with more confidence.

We are doing this as a series of PRs to have easier review and also to incrementally deploy to production.

This one focuses on: 
- common
- some other files connected to common imports

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -40,8 +41,8 @@ type IgnoredTypes = Promise<unknown> | ((...args: unknown[]) => unknown);
 export type ChangeObject<Type> = {
   [Property in keyof Type as Exclude<
     Property,
-    Type[Property] extends IgnoredTypes ? Property : never
-  >]: Type[Property] extends Date ? number : Type[Property];
+    Required<Type>[Property] extends IgnoredTypes ? Property : never
+  >]: Required<Type>[Property] extends Date ? number : Type[Property];
```
Comment (capJavert) [inline src/types.ts:45]: If propery was declared as `date?: Date` in type the `ChangeObject` util would not cast it to `number`. With this change it does so correctly.

File: src/common/object.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@
       return isRequired ? { ...result, [key]: `${key} is required!` } : result;
     }
 
-    const matchResult = value.match(regex);
+    const matchResult = value!.match(regex);
```
[the remark this answers is not in the document]
Comment (capJavert) [reply src/common/object.ts:37]: false positive

**Review by rebelchris — APPROVED** [review]
Looks good :)

File: src/common/pubsub.ts [hunk 3]
```diff
@@ -333,10 +333,10 @@ export const notifyFreeformContentRequested = async (
...
   notifyContentRequested(logger, {
-    id: freeform.payload.after.id,
-    content: freeform.payload.after.content,
-    title: freeform.payload.after.title,
-    post_type: freeform.payload.after.type,
+    id: freeform.payload.after!.id,
+    content: freeform.payload.after!.content,
+    title: freeform.payload.after!.title,
+    post_type: freeform.payload.after!.type,
   });
```
Comment (omBratteng) [inline src/common/pubsub.ts:340]:
```ts
  const after = freeform.payload.after!
  notifyContentRequested(logger, {
    id: after.id,
    content: after.content,
    title: after.title,
    post_type: after.type,
  });
```

Could this work? To not having to do `!` each time?

Comment (capJavert) [reply src/common/pubsub.ts]: yeah, sure 38d60bb170ea246859c834afbe4ecf7a24ed6eac

**Review by omBratteng — APPROVED** [review]

**Pull request merged**